web/Services/AdminService.cs

```diff
@@ -98,46 +98,50 @@ private async Task<List<UserDbModel>> GetPendingAdmins()
         return pendingAdmins;
     }
 
-    public async Task<List<LandlordDbModel>> GetLandlordDisplayList(string approvalStatus)
+    public async Task<List<LandlordDbModel>> GetLandlordList(LandlordListModel landlordListModel)
     {
-        return approvalStatus switch
+        var landlordQuery = _dbContext.Landlords.AsQueryable();
+        if (landlordListModel.IsApproved != null)
         {
-            "Unapproved" => await _dbContext.Landlords.Where(u => u.CharterApproved == false).ToListAsync(),
-            "Approved" => await _dbContext.Landlords.Where(u => u.CharterApproved == true).ToListAsync(),
-            _ => await _dbContext.Landlords.ToListAsync(),
-        };
-    }
-    
-    public async Task<List<TenantDbModel>> GetTenantList()
-    {
-        return await _dbContext.Tenants.ToListAsync();
+            landlordQuery = landlordQuery.Where(l => l.CharterApproved == landlordListModel.IsApproved);
+        }
+        if (landlordListModel.IsAssigned != null)
+        {
+            bool? target = landlordListModel.IsAssigned;

```

Comment:
Do you need this variable?

Reply:
When I tried to write it as a single line, there seemed to be an issue with it being unable to keep track of the various == - I wasn't certain why it was happening, an thought using a separate variable might just help to simplify it slightly.